### ISSUE TITLE
Default Mailaddress in Config for sending Books

### ISSUE THREAD

**Issue by Offerel** [seblucas/cops#336]
As i have seen, the user can specify various parameters in config_local.php which sets defaults for the settings page. Is there a way to specify also the default E-Mail address for sending books to Kindle etc.?

**Comment by hugoheitz** [comment]
In the sendto section of the config file, the address.from is the default email used to send.

* If you want to use a simple smtp server (provided by your ISP for example), you can configure it like that :
     * $config['cops_mail_configuration'] = array( "smtp.host"     => "smtp.free.fr",
     *                                           "smtp.username" => "",
     *                                           "smtp.password" => "",
     *                                           "smtp.secure"   => "",
     *                                           "address.from"  => "cops@slucas.fr",

but maybe you want to define the default destination address ?
it is in the config section of the COPS interface that you can do that for the device used.
by default, it is blank and asks you to fill a value for the email, after that this value will be used for the next emails.

**Comment by Offerel** [comment]
Yes, i mean the default destination address. I know its blank by default, but i would like to have it pre-filled with my kindle address. 

My Problem is, that automatically all my browserdata is cleaned by default and i cant change this. So i would like to have such things saved server based.

**Comment by hugoheitz** [comment]
I understand and I agree with the idea.
The basic would be to add 
"address.to => "destination address by default" ,
in the config file.
And in bookdetail.php to check if there is a value for that and using it by default.
As proposed in #329, I would like to offer the configuration of the destination address in the config file but also the name of the target device so you can easily choose on which device you want to send the content...

**Comment by Offerel** [comment]
This would be great. Also the idea, that you can have more than one destination device.

**Comment by CHBMB** [comment]
I'd add my vote for a selection bar or something similar you can select destinations from.

**Comment by hugoheitz** [comment]
Please vote +1 for  #326 and #329, I describe the functionality and a way to implement it.

**Comment by seblucas** [comment]
Unfortunately voting is easy, implementing it is harder ;)

Anyway I agree that this feature is interesting. I started to hack something based on changing the configuration to something like that : 

```
     * $config['cops_mail_configuration'] = array( "smtp.host"     => "smtp.free.fr",
     *                                           "smtp.username" => "",
     *                                           "smtp.password" => "",
     *                                           "smtp.secure"   => "",
     *                                           "address.from"  => "cops@slucas.fr",
     *                                           "address.to"    => array(
     *                                                                "myKindle1@kindle.com",
     *                                                                "me@gmail.com"
     *                                                              )
     *                                           "subject"       => "Sent by COPS : " // Not mandatory
     *                                           );
```

**Comment by hugoheitz** [comment]
I totally agree, voting is easy, getting it real is harder !

If you manage a family with multiple devices (my situation), the proposition you made to use the mail only is OK to know the correct target if you configure it properly in amazon interface (for the kindles).
You can have something like that 

```

     *                                           "address.to"    => array(
     *                                                                "kindle_dad@kindle.com",
     *                                                                "kindle_mom@kindle.com",
     *                                                                "kindle_son@kindle.com",
     *                                                                "me@gmail.com"
```
(of course, with the correct names).

**Comment by seblucas** [comment]
> How will you merge this evolution and the existing implementation ?

My goal is to open a popup with a checkboxlist of all email adresses I got (coming from the configuration item and from COPS configuration page).

That way you can send to many email at the same time.

> Will there still be a way to modify the destination mail with the config page, I mean this page
> http://cops-demo.slucas.fr/index.php?page=19

You'll be able to add a new email address, yes.

> Do you plan to use by default the default addresses in the config files and as the last item the mail configured in the config page ?

Exactly.

> How can I help you to validate the solution ?

By testing it as soon as I have something ready but that won't be soon (I'll go on vacation in 2 days)

**Comment by hugoheitz** [comment]
Can you use HTML5 to provide something like the following mockup ?
![mockup cops avec with mail button and destination emails list](https://cloud.githubusercontent.com/assets/972023/24765662/e905e5a2-1af8-11e7-94ee-45de64f73969.png)

I am OK to test it as soon as possible but there is no rush. Enjoy your vacation !

**Comment by doug-w** [comment]
Does the kindle built in webbrowser support HTML5?  One of the main
usecases of the mail function to me is to mail from the device to the
device.

**Comment by hugoheitz** [comment]
I dont know for sure about kindle built in webbrowser and the support of HTML5. I proposed that because it is visually better and gives a solution in order to implement the button in the book list page.

I found that information 
https://html5test.com/results/other.html
where the kindle browser is listed...

you are right, if it's important that the function works from the device to the device it needs to work with simple code. a good old select then (as proposed in one of my comments on the issue #329).

keep it simple, it will be easy to debug, if you go into testing the capacity of the web browser, it might bug for unknown reason.

**Comment by CHBMB** [comment]
Yeah, well aware it's easy to say stuff and a hundred times more difficult to implement it.  Sorry if it offended anyone. "Gets my vote" is an English saying.....

**Comment by seblucas** [comment]
@CHBMB 

be sure I was not offended (If I was there would have been no smiley)

@all 

I have something that mostly work, I'll try to create a branch with it soon.

**Comment by hugoheitz** [comment]
Great news ! I am ready to test as soon as it will be ready !

**Comment by CHBMB** [comment]
@seblucas Jolly good, glad to hear I've not upset anyone.  Looking forward to testing this too.

**Comment by hugoheitz** [comment]
Any news about the implementation in a new branch ?
I am ready to test it and I believe it will help to close #336, #329 (if the correct portion of the code is replicated to the page 'main'), #326...

**Comment by hugoheitz** [comment]
After a long time, I would like to know if there is any target date for this set of functionalities...

**Comment by hugoheitz** [comment]
I closed the issue #326 today. 
Any news about this functionalities ?
